### DISCUSSION TITLE
After a system update, some ExplorerPatcher settings are reset.

### DISCUSSION THREAD

**Discussion by septsea** [valinet/ExplorerPatcher#536] [Q&A] [answered]
I am using Windows 11 Dev. I updated my OS today, and I found that some ExplorerPatcher settings had been reset, including:
- Automatically hide the taskbar
- Hide Control Center button
- "When clicking a system icon in the tray, open:" settings
- Maximum number of frequent apps to show
- Remember last used section in this window
- Receive pre-release versions, if available

Perhaps there are more.

Not all the settings are reset, though. For example, I prefer a small taskbar, and the height is not changed after a system update.

The issue can be reproduced. I updated the OS of one of my virtual machines, and the same issue occurred.

**Comment by BraINstinct0** [comment]
Could you provide the version of ExplorerPatcher you upgraded `from` and `to`?
This is needed to first verify if that was mitigated in the latest version.

**Reply by septsea** [reply]
Thanks for your reply.

Well, I didn't upgrade _ExplorerPatcher_ today; I upgraded _Windows 11 Dev_ instead.

The version of EP: 22000.348.40.6.

The version of my OS: from 22509.1011 to 22518.1000.

**Reply by valinet** [reply]
Idk, it doesn't necessarily have to EP that resets those settings. Those are mostly saved in `HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher`, which is a subkey in the key where Explorer stores its settings. Maybe Explorer reset its settings for whatever reason and deleted that key, idk. I don't test on insider builds though, it could very well be a bug or an intended reset by Microsoft as well. Try to collect some more clues and maybe you figure it out, EP doesn't have code to reset its settings randomly, I don't remember for it to ever delete the key.

**Reply by septsea** [reply]
@valinet Thanks for your reply.

Maybe I could do more experiments later to get more clues. I should have made copies of my VMs!

**Comment by septsea** [comment]
The problem persists.

Before upgrading Windows 11 Dev (from 22518.1012 to 22523.1000), I made a copy of my virtual machine so that I could performed a controlled experiment. I installed the latest ExplorerPatcher in Virtual Machine 0, and I did not install EP in VM1. I set two taskbars (in two VMs) hidden automatically. Then I commenced to upgrade Windows 11 Dev. After the update was done, the taskbar in VM0 was visible (in other words, the setting "Automatically hide the taskbar" was reset again), and the taskbar in VM1 was still hidden automatically. You told me to "try to collect some more clues", but I do not know how. I have not figured it out yet.

Maybe I should make a backup of my data and reinstall Windows 11 Stable instead. I do not have spare time to do that until the holiday though. Well, whatever.

**Reply by valinet** [reply]
But besides that, any other setting was reset? Try setting auto hide via Windows’ settings. Does that stick?

I don’t use auto hide, and also use stable Windows builds, but I never experience this, my settings stay as I left them. 

Windows is at liberty to reset whatever it feels like resetting in the registry. It may be that. Also, I do not test Insider builds and do not aim to support them, they’re too volatile and there’s too much work involved for this kind of utility. So yeah, idk, may be a thing happening only on newer builds, may be a bug, idk…

I don’t have specific advice either, sorry.

**Reply by septsea** [reply]
Thanks for your reply.

There are some other settings reset, too. The Control Center button (on the Windows 10 styled taskbar) was visible again; I hid the button before the system update. "When clicking a system icon in the tray, open:" settings were reset as well.

The problem does not stick, because I can hide the taskbar again without any errors.

I also think that insider builds are too volatile; I understand and respect your choice. If you do not plan to test insider builds of Windows 11, I will test stable builds in a month to see whether the issue will persist.

**Reply by valinet** [reply]
The thing is, I don’t know what could be causing this. The only thing I can think of is if Windows resets settings in HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer. EP stores its settings in the ExplorerPatcher subkey in there. That’s the only thing I can think of. To check, uninstall EP, upgrade the build, then verify if the key is still there and populated with anything. If it’s gone or contains nothing, then it means Windows resets it which would suck big time. But at least we’d know and then yeah, the solution would be to get the settings out of there.

The original reason they are in there is because UWP apps cannot read from the registry freely, and way back when I started development, I realized StartMenuExperienceHost.exe can read from there, it fit fine with the theme of the program, so decided to store them there. And initially there were way less entries. Now, a million years later, we are in this situation where the settings are stored in this atypical location…

**Reply by septsea** [reply]
You mean, uninstalling EP before upgrading Windows 11 Dev? I made an extra copy, so I could do another experiment.

**Comment by septsea** [comment]
Your guess is correct.

I uninstalled EP before upgrading Windows 11 Dev, and I could see the subkey in the registry editor:
![image](https://user-images.githubusercontent.com/57337516/146340510-7a406484-5ccd-4c48-ad8c-ef9b4772ba49.png)

After the update, it is nowhere to be seen.
![image](https://user-images.githubusercontent.com/57337516/146340572-e899436f-1b77-4763-bdec-f8302656bcd4.png)

**Reply by valinet** [reply]
Okay. Well, that kind of sucks. The best solution would be to move the settings some place else and find a registry key that would work for `StartMenuExperienceHost.exe` as well. Kind of sucks, but whatever... I should implement that at some point. It takes a fair bit of refactoring. Too bad Windows does this, I don't understand why it feels the need to do this.

**Comment by valinet** [comment, marked as answer]
Yeah, so I promoted this to an issue, it should be fixed somehow in a subsequent update at some point. Track it [here](https://github.com/valinet/ExplorerPatcher/issues/576).

Thanks for the report and investigation @septsea

**Reply by septsea** [reply]
Thanks for fixing the bug.
